### PR TITLE
Add generation of interpolatable masters

### LINKED ISSUE #471: Provide interpolatable TTFs as part of releases
For merging Google Sans into Google Sans Arabic (which will remain a separate family), we merge TTF masters and build variable fonts from them instead of doing a source merge with all the complications of differences in build tooling.

To do that, we need interpolatable TTFs for Google Sans. For now, I built them with manually with:
```
fontmake -m source/GoogleSans/GoogleSans.designspace -o ttf-interpolatable --output-dir build/GoogleSans/ttf-interpolatable
```

But it would be nice if there were a make target for this and the TTFs be part of the release (may be in a separate tarball), so that updating the Latin part of the Arabic font is as streamlined as possible.

(P.S. I tried instantiating masters from the variable font, and it worked for most part until I found out that quite a few glyphs ended up not being compatible after instantiation and varLib was ignoring them).

### PR DESCRIPTION
**Pull request by madig**
Add a Makefile target to generate interpolatable masters.

Should it be added to `all:`? 

Closes #471

### CODE REVIEW

**Comment by chrissimpkins** [comment]
> Should it be added to all:?

Does it need to be in the release archive?  If so, we should.  It looks like the GHA release workflow uses `make -j` to build the release.

**Comment by chrissimpkins** [comment]
cc @khaledhosny re: #471 request

File: source/Makefile [hunk 1]
```diff
@@ -137,6 +138,14 @@ VARIABLE_BUILD_PATH_ITALIC_TARGET: $(FONT_BUILD_DIR)
 	font-v write --sha1 $(VARIABLE_BUILD_PATH_ITALIC)
 	mv $(VARIABLE_BUILD_PATH_ITALIC) "$(VARIABLE_BUILD_PATH_ITALIC_TARGET)"
 
+gs-compatible-masters: gs-compatible-masters-upright gs-compatible-masters-italic
+
+gs-compatible-masters-upright: $(FONT_BUILD_DIR)
+	$(FONTMAKE_EXE) -m $(UPRIGHT_SOURCE) -o ttf-interpolatable --output-dir $(INTERPOLATABLE_BUILD_DIR) --expand-features-to-instances
```
Comment (khaledhosny) [inline source/Makefile]: What is the purpose of `--expand-features-to-instances` here? (I have been building without it, so I hope I’m not missing something).

**Comment by khaledhosny** [comment]
> Does it need to be in the release archive?

That would be ideal for me, so that there is a canonical source of these files.

File: source/Makefile (hunk 1 above)
Comment (madig) [reply source/Makefile]: Oh, this is primarily interesting for instance generation, I think I made a copy-paste oopsie.

**Comment by chrissimpkins** [comment]
Sgtm.  Let's add it to the default all target and include in the release archives.  

@madig how will this work with the transition to statics instanced from vf’s targets?  

And should we add new CI workflows to cover the new build targets? This is a bigger question that influences our expected values for HB regression testing, expected glyph name/order, likey other QA

**Comment by madig** [comment]
They are separate from all! If anything, I could base the VF builds off the interpolatable builds but will have to brush up on my Makefile-fu to figure that one out...

I don't think we need dedicated QA targets, as long as the command to generate them is the same as the one used to generate VFs.

**Comment by chrissimpkins** [comment]
> If anything, I could base the VF builds off the interpolatable builds 

If this means a transition to a different approach to the vf format builds, let's not do that now.  We have a number of things to address and don't need another extensive binary diff review.

**Comment by madig** [comment]
Aaaargh. I tried to base VF generation on ttf-interpolatable generation just to see what happens, and there are indeed binary diffs. A very cursory look suggests they are mainly reordering things but I can't be sure. Let's go with separate build steps then...

**Comment by chrissimpkins** [comment]
We should land this and release a new v8.001 to support downstream family builds

**Comment by chrissimpkins** [comment]
@khaledhosny We plan to push a new v8.001 release with the new build format this week.  Do things in these builds look ok for your use?

**Comment by khaledhosny** [comment]
Is there a way to download the files built in this PR to double check locally? Otherwise the change looks good and should produce similar files to what I have been manually building.

**Comment by madig** [comment]
We're not uploading artifacts from the build jobs themselves, so you'll have to run it locally...

**Comment by khaledhosny** [comment]
I’m not sure if this is expected or not:
```
$ make gs-compatible-masters-upright
make: *** No rule to make target `gs-compatible-masters-upright'.  Stop.
$ make gs-compatible-masters-italic
make: *** No rule to make target `gs-compatible-masters-italic'.  Stop.
```

`make gs-compatible-masters` works, though. The files look fine, but they use production glyph names and the files I was building manually use source glyph names. I think source glyph names make more sense here since these are not the final production files.

**Comment by chrissimpkins** [comment]
> I’m not sure if this is expected or not:
> 
> ```
> $ make gs-compatible-masters-upright
> make: *** No rule to make target `gs-compatible-masters-upright'.  Stop.
> $ make gs-compatible-masters-italic
> make: *** No rule to make target `gs-compatible-masters-italic'.  Stop.
> ```
> 
> `make gs-compatible-masters` works, though. The files look fine, but they use production glyph names and the files I was building manually use source glyph names. I think source glyph names make more sense here since these are not the final production files.

Thoughts @madig?

**Comment by madig** [comment]
Added top-level targets and keep design names.

File: source/Makefile [hunk 2]
```diff
@@ -38,6 +38,7 @@ ITALIC_SOURCE=GoogleSans/GoogleSans-Italic.designspace
 FONT_BUILD_DIR=../build/GoogleSans
 STATIC_BUILD_DIR=$(FONT_BUILD_DIR)/static
 VARIABLE_BUILD_DIR=$(FONT_BUILD_DIR)/variable
+INTERPOLATABLE_BUILD_DIR=$(FONT_BUILD_DIR)/interpolatable
```
Comment (chrissimpkins) [inline source/Makefile:41]: For future reference, this PR will add a new set of ttf-interpolatable artifacts on the path `build/GoogleSans/interpolatable`.  We need these builds to support downstream G Fonts Latin Core glyph set merges into script-specific/style-specific GS families.

**Review by chrissimpkins — APPROVED** [review]

**Comment by chrissimpkins** [comment]
Can confirm that our release.yml workflow uses `make -j` to compile the release archive.  We should include the new build format in all future releases.

**Comment by chrissimpkins** [comment]
Running commit https://github.com/googlefonts/googlesans/pull/496/commits/917d1316d20d3276fad92974312d102726abedf1 on a build-* branch to execute the full suite of CI tests.

File: source/Makefile (hunk 2 above)
Comment (chrissimpkins) [reply source/Makefile:41]: These artifacts will be released in the git tagged GH Releases archives as of v8.001

**Comment by chrissimpkins** [comment]
Merging so that we can release v8.001.

@madig is it helpful to include a workflow that will add these builds to downloadable archives that are pushed with commits on our build-* branches?  Or an extra branch / tag targeted build process that allows someone to receive the new format compiled from pre-release sources?

**Comment by madig** [comment]
I just thought that maybe we can drop GPOS compression on interpolatable builds because why compress an intermediate artifact.

Hm... not sure. I think you typically want just the release package?

**Comment by khaledhosny** [comment]
I think I tried with and without GPOS compression and saw no difference in the final Google Sans Arabic files.

**Comment by khaledhosny** [comment]
(I mean building Google Sans masters w/o compression made no difference in my final Google Sans Arabic files)

**Pull request merged**